Pod/Classes/Exp.swift

```diff
@@ -407,26 +409,38 @@ public func findFeeds(params:[String:AnyObject]) -> Promise<SearchResults<Feed>>
 }
 
 
+
+
 /**
-Login EXP system
-@param user,password,organization.
-@return Promise<Token>.
-*/
-func login(user:String,passwd:String,organization:String) ->Promise<Token>{
+ Login EXP system
+ @param user,password,organization.
+ @return Promise<Token>.
+ */
+func login(options:[String:String]) ->Promise<Auth>{
     return Promise { fulfill, reject in
-        let req = Alamofire.request(Router.login(["username":user,"password":passwd,"org":organization]))
-            .responseObject { (response: Response<Token, NSError>) in
+        Alamofire.request(Router.login(options))
+            .responseObject { (response: Response<Auth, NSError>) in
                 switch response.result{
                 case .Success(let data):
                     fulfill(data)
+                    let expiration = data.get("expiration") as! Double
+                    let startDate = NSDate(timeIntervalSince1970: expiration/1000)
+                    let timeout = startDate.timeIntervalSinceDate(NSDate())
+                    after(NSTimeInterval(Int64(timeout))).then{ result -> Void in
+                        runtime.start(runtime.optionsRuntime)
```

Comment:
instead of calling start() again this should probably only call the refresh end point `/api/auth/token` so it doesn't affect the channels